superset-frontend/plugins/legacy-plugin-chart-event-flow/src/controlPanel.tsx

```diff
@@ -46,10 +47,8 @@ const config: ControlPanelConfig = {
                 choices: columnChoices(state?.datasource),
               }),
               // choices is from `mapStateToProps`
-              default: (control: { choices?: string[] }) =>
-                control.choices && control.choices.length > 0
-                  ? control.choices[0][0]
-                  : null,
+              default: (control: ControlState) =>
+                control.choices?.[0]?.[0] || null,
```

Comment:
How about `ensureIsArray(control.choices)[0][0]`? I'm actually not sure which one is easier to read, up to you 🙂

Reply:
Wouldn't it need to be `ensureIsArray(control.choices)[0]?.[0]` ? If so, I think the current one is probably ok (mind you, I agree it's pretty ugly 😄 )